### ISSUE TITLE
Add Android TV support on the server side

### ISSUE THREAD

**Issue by torshie** [MayaPosch/NymphCast#22]
With Android TV support on the server side, we can enjoy all the goodies of NymphCast without the burden (cable hazard, space, money ) of an extra device.

**Comment by MayaPosch** [comment]
What do you mean exactly with 'Android TV support' in this context? Have the NymphCast server implemented as an Android TV app?

**Comment by torshie** [comment]
@MayaPosch Yes. I'm not sure whether it's possible to do that, considering all those restrictions on Android TV.

**Comment by MayaPosch** [comment]
A quick look shows that SDL2 is supported on Android, as is ffmpeg. This might be a realistic proposition :)

I guess it would be nice to have a local GUI for NymphCast-on-Android-TV as well. This is something that might be generally useful, even outside of Android TV. A more Roku-like experience, basically.

It appears that Android TV apps are standard Android apps, looking at the documentation. This should mean that one can test it on any Android device before deploying it on an Android TV device, if I'm not mistaken.

**Comment by torshie** [comment]
Yes, Android TV apps are standard Android apps, in fact you can install and run many Android apks directly on an Android TV, e.g. a browser app.
However TV apps are usually controlled with a remote, while phone apps are controlled with a touch screen, so the UI part may need some work.

**Comment by MayaPosch** [comment]
To implement Android TV support, it appears that the easiest way might be to create a NymphCast client app that communicates with an (unmodified) NymphCast Server instance that runs as a background service. Because the client app would launch the background service, this would prevent the service from being killed by the OS.

Exact implementation details and maybe other options need some more brainstorming on, I think.

**Comment by fransvanberckel** [comment]
I searched for, how this is done by Chromecast v2, on Android TV. They are using the com.google.cast.player.message and com.google.cast.media libraries on the server side. These services are always running. So no need to start a app. It's well documented, now ...

https://github.com/thibauts/node-castv2

**Comment by SpiderUnderUrBed** [comment]
Any updates?

**Comment by MayaPosch** [comment]
@SpiderUnderUrBed I have created a version of NymphCast Server that runs on Android, but after submitting it to the Google Play Store it reported a crashing issue on x86_64 Android that I have been unable to replicate so far. On AArch64 Android it seemed reasonably stable.

For the upcoming v0.2-alpha0 pre-release I'll add an APK to the pre-release page, and there are build instructions in the documentation folder for those who want to try. I'll add more detailed instructions as soon as this stabilises somewhat.

**Comment by SpiderUnderUrBed** [comment]
An apk to the pre-release page will be nice, but the app isnt appearing on the play store on android tv. It would be difficult to get this working if you had to download the apk, put it on a usb, then try and open it on the tv. But doable. 

I'll be sure to try once  v0.2 comes out.

**Comment by MayaPosch** [comment]
Yeah, it's not available yet on the store due to the x86_64 stability issue. I recall being able to replicate it with an x86_64 Android VM identical to the one that their automatic testing system uses, but haven't found the time yet to debug and fix it. Since NCS runs fine on x86_64 Linux and Windows, it's a bit of a weird issue.

I might just put an AArch64-only version on the Play Store to bypass it, in addition to the APK on the pre-release page.
